gamemodes/cinema/gamemode/modules/theater/services/sh_youtube.lua

```diff
@@ -30,22 +30,125 @@ function SERVICE:GetKey(url)
+	]]
+	local loaded = false
+	local function RandomString()
+		local s = {}
+		for i=1,10 do
+			s[i] = string.char(math.random(65,90))
+		end
+		return table.concat(s)
+	end
+	
     function SERVICE:LoadVideo(Video, panel)
         panel:EnsureURL("http://swamp.sv/s/cinema/" .. self:GetClass() .. ".html")
-        -- Let the webpage handle loading a video
-        local str = string.format("th_video('%s',%s);", string.JavascriptSafe(Video:Key()), Video:Duration() > 0 and "false" or "true")
-        panel:QueueJavascript(str)
-        local fn = panel.ConsoleMessage
-
+		panel.vid = Video
+		local enabled = false
+		
+		if (self:IsMature(Video)) then
+			panel:EnsureURL("https://www.youtube.com/embed/"..Video:Key().."?autoplay=1&controls=0&showinfo=0&modestbranding=1&rel=0&iv_load_policy=3&enablejsapi=1")
+			enabled = true
+			loaded = false
+			timer.Create(RandomString(), .1, 30, function()
+				if IsValid(panel) then
+					if enabled then
+						panel:RunJavascript(payload)
+						panel:RunJavascript("if(window.ytcfg)console.log('PAGE LOADED');")
+						self:SetVolume(theater.GetVolume(), panel)
+					end
+					panel:RunJavascript([[
```

Comment:
This doesn't belong here, it belongs in the html file.

Reply:
the whole string of javascript fails if there's a single error so I need them in separate strings, also there's not really any difference between using a webpage and using sethtml except for latency however in this case I _need_ to have it open the youtube embed page due to web protection crap like CORS

Reply:
Oh I see, I didn't notice that you were changing the page URL. This is good then.